File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    let stack = [];
   
    for(let i=0; i<ingredient.length; i++){
        stack.push(ingredient[i]);
        
        if(stack.length >= 4 && stack.slice(-4).join('') === '1231'){
            answer++;
            stack.splice(-4);
        }
    }
    
    return answer;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        int[] burger = {1,2,3,1};
        
        Stack<Integer> stack = new Stack<>();
        
        for(int i : ingredient){
            stack.push(i);
            
            if(stack.size()>=burger.length &&
               stack.get(stack.size() - 4) == burger[0] && 
               stack.get(stack.size() - 3) == burger[1] && 
               stack.get(stack.size() - 2) == burger[2] && 
               stack.get(stack.size() - 1) == burger[3]){
                
                stack.pop();
                stack.pop();
                stack.pop();
                stack.pop();
                answer++;
                
            }
        }
        return answer;
    }
}